src/python/pants/backend/python/tasks/resolve_requirements.py

```diff
@@ -24,7 +25,28 @@ def product_types(cls):
   def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
 
+  @classmethod
+  def register_options(cls, register):
+    super(ResolveRequirements, cls).register_options(register)
+
+    register('--force', type=bool, default=False, fingerprint=True,
+             help='Always run requirements resolution, even if all the target roots are '
+                  'python_binary()s.')
+
   def execute(self):
+    # TODO: PythonBinaryCreate resolves its own requirements separately, more sensitively to
+    # platform constraints. No-oping here is a bit of a hack and will cause repls on python_binary()
+    # targets to fail.
+    if all(PythonBinaryCreate.is_binary(t) for t in self.context.target_roots):
+      if self.get_options().force:
+        self.context.log.info('Requirements pex task run with --force.')
+      else:
+        self.context.log.info(
+          'Requirements pex was not generated when target roots are only python_binary() targets. '
```

Comment:
I like that you are being conservative here, in case it fails to run `repl`although, I would change this to a `debug` log and reduce to something like:

'./pants repl is not guaranteed to succeed because of requirements resolution, re-run with --force'

Reply:
I think it would also break other goals like `./pants run`, and since I am concerned that this may immediately break some scripts (with an easy fix, but still!) I feel like I would want to turn this into a `warn` log -- just something that immediately pops up saying what's failing and how to fix it. Does that sound reasonable?

Even given that, I would love to reduce the amount of density in the message (it's also prone to bitrot) so maybe a link to the ticket would be better than writing it out here?

I think making the `Requirements pex task run with --force.` log into `debug` is probably more correct, so I will make that change.

Reply:
I think this applies here:

> One of my main confusions is that a PEX can be used in many situations other than pants.pex, and the word "goal" throws me off because goals only apply to pants.pex.

Can't this PEX be used for things other than `./pants`, and if that's the case then it doesn't make sense to always refer to goals like `repl` and `run`.